src/codegen/templates/_grpc_interpreter.py.mako

```diff
@@ -137,10 +139,23 @@ class GrpcStubInterpreter(BaseInterpreter):
         %endif
         %if is_read_method:
         % for param in get_read_array_parameters(func):
-        ${param}[:] = response.${param}
+        _assign_numpy_array(${param}, response.${param})
         % endfor
         %endif
         %if len(output_parameters)  > 0:
         return ${get_response_parameters(func)}
         %endif
-% endfor
\ No newline at end of file
+% endfor
+
+def _assign_numpy_array(numpy_array, grpc_array):
+    """ Assigns the grpc array to the numpy array, while still maintaining the 
+    original shape of the numpy array. 
+    
+    The numpy array assignment is skipped when the length of the grpc array 
+    is greater than the numpy array as this might result in data loss.
+    """
+    grpc_array_size = len(grpc_array)
+    if numpy_array.size <= grpc_array_size:
+        numpy_array.flat[:grpc_array_size] = grpc_array
+    else:
+        raise errors.DaqError(error_codes.DAQmxErrors.READ_BUFFER_TOO_SMALL)
```

Comment:
DaqError needs 2 parameters. You need to specify an error message.

Further investigation reveals that https://github.com/ni/nidaqmx-python/blob/96ab6c1679acf02a89999a5124fc8e029620a2ff/generated/nidaqmx/stream_readers.py#L83 has some examples of read errors thrown from Python.

```suggestion
        raise errors.DaqError(
            'Read cannot be performed because the NumPy array passed into '
            'this function is too small. You must pass in a NumPy array of the '
            'correct size based on the number of channels in the task and the '
            'number of samples per channel requested.\n\n'
            'NumPy array size: {}\n'
            'Required size: {}'
            .format(numpy_array.size, grpc_array_size),
            error_codes.DAQmxErrors.READ_BUFFER_TOO_SMALL)
```

Reply:
updated as per suggestion